docs/TemplateManagementCLI.md

```diff
@@ -0,0 +1,100 @@
+
+* Azure Active Directory service principal (optional) - If using service principal's identity for authentication, you need to create a [service principal](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-auth-service-principal) to access your registry. Ensure that the service principal is assigned a role such as AcrPush so that it has permissions to push and pull artifacts.
+
+* Azure CLI (optional) - To use an individual identity, you need a local installation of the Azure CLI. Version 2.0.71 or later is recommended. Run `az --version` to find the version. If you need to install or upgrade, see [Install Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli).
+
+* Docker (optional) - To use an individual identity, you must also have Docker installed locally, to authenticate with the registry. Docker provides packages that easily configure Docker on any [macOS](https://docs.docker.com/docker-for-mac/), [Windows](https://docs.docker.com/docker-for-windows/), or [Linux](https://docs.docker.com/engine/install/) system.
+## Authentication
+
+Before pull & push operations, azure authentication is required for private registries. Customers can directly use individual login with Azure AD through [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/authenticate-azure-cli) or use identity (individual identity or Azure AD [service principal identity](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-auth-service-principal)) to sign in the registry. 
+
+### Login Using Azure CLI
+
+After signing in to the Azure CLI with your identity, use the Azure CLI command `az acr login` to access the registry.
+```
+> az acr login --name <registry>
+```
+
+### Login Using Identity (individual or service principal indentity)
+
+* Docker login
+
+```
+> docker login <registry> -u <username> -p <password>
+```
+* Oras Login
+
+The [oras](https://github.com/deislabs/oras) tool oras.exe is packed in our repo, users can directly use it for login as follows.
+
+```
+>.\oras.exe login <registry> -u <username> -p <password>
+```
+
+## Push
+To push a template collection, the command is: 
+
+```
+push <ImageReference> InputTemplateFolder [ -n | --NewBaseLayer]
+```
+| Value | index |Optionality |  Description |
+| ----- | ----- | ----- |----- |
+| ImageReference |0| Required |  Image reference: \<Registry>\/\<Name> \[:\<Tag>]  (Image name should be lowcase.)|
+|InputTemplateFolder | 1 |Required |Input template folder. |
+
+| Option | Name | Optionality | Default | Description |
+| ----- | ----- | ----- |----- |----- |
+| -n | BuildNewBaseLayer | Optional | false | Ignore previous base layer and build new layer. |
+
+Example command to push a collection of templates to ACR image from a folder:
+
+```
+>.\Microsoft.Health.Fhir.Liquid.Converter.Tool.exe push testacr.azurecr.io/templatetest:default myInputFolder
+```
```

Comment:
If there some constraint on the input folder? Should the entry template be present directly in the root folder, or it can be in any subfolder too?

Reply:
Yes, the entry template should be present directly in the root folder.
Added a note in the doc.